src/foam/nanos/notification/email/DAOEmailService.js

```diff
@@ -92,55 +112,173 @@ return config_;`
+        if ( SafetyUtil.isEmpty(emailMessage.getReplyTo()) ) {
+          if ( ! foam.util.SafetyUtil.isEmpty(emailTemplate.getReplyTo()) ) {
+            JtwigTemplate templateDisplayName = JtwigTemplate.inlineTemplate(emailTemplate.getReplyTo(), config);
+            emailMessage.setReplyTo(templateDisplayName.render(model));
+          } else {
+            emailMessage.setReplyTo(
+              ! SafetyUtil.isEmpty(group.getReplyTo()) ?
+                group.getReplyTo() : getReplyTo()
+            );
+          }
+        }
+
+        // DISPLAY NAME:
+        if ( SafetyUtil.isEmpty(emailMessage.getDisplayName()) ) {
+          if ( ! foam.util.SafetyUtil.isEmpty(emailTemplate.getDisplayName()) ) {
+            JtwigTemplate templateDisplayName = JtwigTemplate.inlineTemplate(emailTemplate.getDisplayName(), config);
+            emailMessage.setDisplayName(templateDisplayName.render(model));
+          } else {
+            emailMessage.setDisplayName(
+              ! SafetyUtil.isEmpty(group.getDisplayName()) ? group.getDisplayName() : getDisplayName()
+            );
+          }
+        }
+
+        // SUBJECT:
+        // Since subject is very specific to each email there is no group field or default value for this property.
+        if ( foam.util.SafetyUtil.isEmpty(emailMessage.getSubject()) &&
+          ! foam.util.SafetyUtil.isEmpty(emailTemplate.getSubject()) ) {
+            JtwigTemplate templateSubject = JtwigTemplate.inlineTemplate(emailTemplate.getSubject(), config);
+            emailMessage.setSubject(templateSubject.render(model));
+        }
+
+        // SEND TO:
+        // Since sendTo is very specific to each email there is no group field or default value for this property.
+        if ( emailMessage.getTo().length == 0 &&
+          ! foam.util.SafetyUtil.isEmpty(emailTemplate.getSendTo()) ) {
+            JtwigTemplate templateSendTo= JtwigTemplate.inlineTemplate(emailTemplate.getSendTo(), config);
+            emailMessage.setTo(new String[] {templateSendTo.render(model)});
+        }
+
+        return emailMessage;
+      `
+    },
+    {
+      name: 'findUser',
+      type: 'foam.nanos.auth.User',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'emailMessage',
+          javaType: 'final foam.nanos.notification.email.EmailMessage'
+        }
+      ],
+      javaCode:
+        `
+        foam.nanos.session.Session session = x.get(foam.nanos.session.Session.class);
+
+        DAO userDAO         = (DAO) x.get("localUserDAO");
+        User user           = (User) userDAO.find(session.getUserId());
+
+        // 1. If the user doesn't login at this time, get the user from localUserDao
+        // 2. If the user is the system user, get the real user from localUserDao
+        if ( user == null || user.getId() == 1 ) {
+
+          Sink sink = new ArraySink();
+          sink = userDAO.where(MLang.EQ(User.EMAIL, emailMessage.getTo()[0]))
```

Comment:
Missing `.inX(x)`

Reply:
inX() makes sense when creating/calling the dao, however not after, from my understanding.

from: foam2/doc/guides/StyleGuide.md
##### When using a DAO as a regular client from outside of a decorator for that DAO, use the non-CO methods and use inX() to specify the context. ie.
```javascript
   DAO userDAO = ((DAO) x.get("userDAO")).inX(x);
   userDAO.find(id);
```

Reply:
It makes no difference where you put it. If you want to put it where you define the variable then you don't need to put it at call sites and vice versa.